src/library/scala/concurrent/duration/Duration.scala

```diff
@@ -124,6 +124,8 @@ object Duration {
       Undefined
     else if (nanos > Long.MaxValue || nanos < Long.MinValue)
       throw new IllegalArgumentException("trying to construct too large duration with " + nanos + "ns")
+    else if (nanos < 0)
+      fromNanos((nanos - 0.5).toLong)
```

Comment:
`toLong` truncates the decimal part. The `+ 0.5` part is meant to round the decimal number to the nearest integer number. `Duration(0.5, NANOSECONDS) == Duration(1, NANOSECONDS)` and `Duration(0.4, NANOSECONDS) == Duration(0, NANOSECONDS)`. Not sure though why using Double over Long is necessary.